hw/opentitan/ot_dma.c

```diff
@@ -963,6 +968,24 @@ static void ot_dma_abort(OtDMAState *s)
     timer_mod(s->timer, (int64_t)now);
 }
 
+static void ot_dma_trigger_fatal_error(OtDMAState *s, const char *func,
+                                       int line)
+{
+    const OtDMAOp *op = &s->op;
+
+    error_report("%s: %s: DMA fatal error @ line %d", func, s->ot_id, line);
+    error_report("  %s addr 0x%" PRIx64 " size 0x%" PRIx64 " chunk %u",
+                 op->write ? "WR" : "RD", op->addr, op->size, op->chunk);
+    error_report("  asix %s mr %s (%p)", AS_NAME(op->asix),
+                 op->mr ? op->mr->name : "null", op->mr);
+#ifdef MEMTXATTRS_HAS_ROLE
```

Comment:
Should maybe be
```suggestion
#ifdef OT_DMA_HAS_ROLE
```
instead?

Reply:
The rationale behing this very weird definition is the following:

Rivos has added a new bitfield to MemTxAttrs which is very Rivos-specific. So the Rivos version has an extra field, which does not appear in the upstream QEMU version - and is useless for OT, at least up to now.

In order to share code without breaking builds, I've asked to add this `MEMTXATTRS_HAS_ROLE` which is not specific to DMA nor to OT, but to Rivos. If not present, the code is ignored by the compiler. If it is, then the bitfield is present and accessed.

Note that other companies (SiFive) have also added custom fields to this MemTxAttrs structure (for very similar same reason). There also some platform-specific, pre-existent, unusable bits in this structure, such as the ARM-TrustZone `secure` bit - which also serve a similar purpose. At some point, it would be very nice that QEMU upsteam defines platform-agnostic reserved bits in this structure to pass along this kind of information, unfortunately it does not seem to be priority.

Getting back to the comment, I cannot change thed definition name which is not related to the DMA, but to the whole machine, and tells whether this is a `role` bitfield within the `MemTxAttrs` structure. I hope it clarifies the somewhat surprising definition.

Reply:
That makes sense, can you maybe add a comment somewhere about this, just to the effect of "check `MEMTX_ATTRS_HAS_ROLE` instead of `OT_DMA_HAS_ROLE` to support common downstream extensions to `MemTxAttrs`" or similar?

Reply:
Note: the extra field we have added is for RACL role. Since RACL is now part of Darjeeling, the relevant MemTxAttrs code will be contributed to this repo at some point.

Reply:
Second though: I can use OT_DMA_HAS_ROLE here. I've added a comment where we define `OT_DMA_HAS_ROLE` from `MEMTXATTRS_HAS_ROLE`.